clang/test/CheckedCRewriter/alloc_type_param.c

```diff
@@ -28,15 +28,14 @@ void foo() {
 
 // Allocating pointers to pointers
 void bar() {
-  // The type parameter doesn't need to be _Ptr<int> even though it is checked
   int **a = malloc(sizeof(int*));
-  // CHECK: _Ptr<_Ptr<int>> a = malloc<int *>(sizeof(int*));
+  // CHECK: _Ptr<_Ptr<int>> a = malloc<_Ptr<int>>(sizeof(int*));
   *a = malloc(sizeof(int));
   // CHECK: *a = malloc<int>(sizeof(int));
 
   // It's also fine if the pointer is unchecked
   int **b = malloc(sizeof(int*));
-  // CHECK: _Ptr<int*> b = malloc<int *>(sizeof(int*));
+  // CHECK: _Ptr<int*> b = malloc<int*>(sizeof(int*));
```

Comment:
I see this change a lot -- can you add back the space so it's `int *` and not `int*` ? I think the former is more typical in style guides so we are less likely to break peoples' formatting in their code.

Reply:
I spent some time trying to keep the space earlier, but adding the space there means that `_Ptr<int*>` is output as `_Ptr<int *>` everywhere. There wasn't a space in the previous PR because the string was generated by a clang builtin `toString` method. With this PR, the string comes from our `mkString` method in `ConstraintVariable`.

Reply:
I'm a little confused. I think you are saying that if you fix the rewriting code to handle this case, it will change rewriting for other cases, so it's a wash?  

You are saying the clang builtin doesn't add the space? Is the clang builtin actually a Checked C extension to the clang code, and not clang's?

Reply:
It's a clang thing. `QualType` has the method `getAsString` that returns `int *`.  The way we convert ConstraintVariables to strings does not add a space (i.e., it returns `int*`).

I'm starting to think we should modify our `mkString` method to add this space in order to be consistent with clangs output. I might make this is a separate PR, since I think it will affect a significant number of tests and is mostly orthogonal to the changes in this PR.

Reply:
That’s a good idea. Please do that.

Reply:
So: Looks we are still waiting on this?